### PR TITLE
added chef server install graph

### PR DESCRIPTION
**Pull request by VulpesArtificem**
work in progress for part of https://rackhd.atlassian.net/browse/RI-136
This graph can install and prepare a chef server on a node and subsequently another graph can be run w/ the proper credentials to install the chef client on other nodes and register them with the server. However, at the moment there's not a great place to put the various keys created during the chef-server process so the key-fetching portion isn't automated and if we're going to continue integrating with other services that want some key(s) to be associated with a node it might be a good idea to talk about some kind of database or other encrypted key storage/fetching.
@pscharla @RackHD/corecommitters

### CODE REVIEW

**Comment by PengTian0** [comment]
test this please

**Comment by pscharla** [comment]
@VulpesArtificem : for your above question related to where to store the keys, lets add that to the discussion list for the CC meeting.

**Comment by VulpesArtificem** [comment]
@pscharla is that not architecture?

**Comment by pscharla** [comment]
@VulpesArtificem : yeah - you could start there and then delve into details in the CC meeting if necessary. So I'll let you go ahead and get it on the agenda for the next architecture meeting

File: spec/lib/graphs/install-chef-server-graph-spec.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+// Copyright 2017, EMC, Inc.
```
Comment (pscharla) [inline spec/lib/graphs/install-chef-server-graph-spec.js]: @VulpesArtificem : Updated copyright needed here.

**Comment by pscharla** [comment]
jenkins: test this please

**Comment by pscharla** [comment]
jenkins: test this please

**Comment by pscharla** [comment]
jenkins: test this please

**Comment by pscharla** [comment]
@RackHD/corecommitters please review this PR for us.

**Comment by pscharla** [comment]
@anhou can you take a look at this one?  Since you reviewed the other graph for the Chef client. This is the workflow for installing a Chef server on a node.

**Comment by lacarb** [comment]
+1 looks good

**Comment by pscharla** [comment]
jenkins: test this please

**Comment by pscharla** [comment]
:+1: 
@anhou : this one is ready to go. Can you merge for us assuming there is no freeze?
Thanks - Paul

**Pull request merged**